pyhms/cluster/landscape_approximator.py

```diff
@@ -0,0 +1,42 @@
+import numpy as np
+
+from ..core.problem import get_function_problem
+from ..demes.cma_deme import CMADeme
+from ..demes.single_pop_eas.sea import MWEA
+from ..tree import DemeTree
+from .cluster import Cluster
+from .consolidator import PairwiseNeighborConsolidator
+from .kriging import KrigingLandscapeApproximator
+from .local_basin_agent import LocalBasinAgentExecutor
+from .merge_conditions import MergeCondition
+
+
+class LandscapeApproximator:
+    def __init__(
+        self,
+        hms_tree: DemeTree,
+        merge_condition: MergeCondition,
+        local_basin_epochs: int,
+        mwea: MWEA | None = None,
+    ) -> None:
+        assert all(isinstance(deme, CMADeme) for deme in hms_tree.leaves), "All leaves must be CMADeme instances"
```

Comment:
It is unintuitive from the user's perspective, that "All leaves must be CMADeme instances". Is there a reason behind this other than CMADeme having implemented mean and covariance_matrix properties?

Reply:
The CMADeme is the only one that has covariance_matrix (we need it for LocalOptimizationMergeCondition). The landscape approximation was also tested only with CMA-ES.

Reply:
I understand it can be confusing, but I don't anticipate a high volume of users for this feature. 😄